### ISSUE TITLE
Adjust access modifier on Client class methods

### ISSUE THREAD

**Issue by RobinRadic** [plesk/api-php-lib#5]
The `PleskX\Api\Client` class has a fair amount of methods and properties with a private access modifier. Consider changing it to protected.

I would like to extend the class and adjust a few lines in one of the methods. But currently it's a pain in the butt because most methods/properties are declared as `private`.

**Comment by sibprogrammer** [comment]
Can you show your code or describe the problem with more details? Which particular field or method your trying to override in child class? What for?

**Comment by RobinRadic** [comment]
What i was trying to do, is adding `curl_error` information when `curl_exec` returns false. I wanted more details as to why some requests failed. So i extended the class to override `_performHttpRequest`.
```php
    private function _performHttpRequest($request)
    {
          // code...

        $result = curl_exec($curl);

        $error = false;
        if($result === false){
            $error = curl_error($curl);
        }

        if (self::$_isExecutionsLogEnabled) {
            self::$_executionLog[] = [
                'trace' => debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS),
                'request' => $request,
                'response' => $result,
                'error' => $error
            ];
        }

       // code...
    }
```

But in more general terms, i can think of several scenarios where one would like to alter some functionality inside the `Client`. Currently with all the private methods and properties its not sufficient to extend and override. Have to copy/paste the complete class.

**Comment by sibprogrammer** [comment]
Fixed in fb5edbb17d6052239fa8575350919780a83edcc3 and e541ba33956f31fa7772a1b1e0eaaf23bc8819b2
